### ISSUE TITLE
Loading error when custom date range is too wide

### ISSUE THREAD

**Issue by hafizrahman** [woocommerce/woocommerce-ios#12147]
To replicate issue:

1. Start with site with no Custom Range,
2. Create a new custom range. In the date picker, pick a starting date very long in the past (e.g: in year 1999)
3. Create the custom range tab
4. Notice that error screen is shown, and the stats tab is showing blank info with no option to change date range or fix the situation.
5. Surprisingly, you can also tap on the empty space where the date range label is supposed to be, and it will show the date range picker.

See video:

https://github.com/woocommerce/woocommerce-ios/assets/266376/9fb7d51d-4f6c-46a3-b1a2-68caa74bc00a

A quick fix might be to make the date range label on step 5 visible.

Also notice that if you change the date to be something that doesn't trigger the "we couldn't load your data" error panel, it will load the data correctly but doesn't dismiss the "we couldn't load your data" error panel.

---

Upon further investigation, it seems like the problem with loading failure is due to `rest_invalid_param` for `per_page` parameter passed into `OrderStatsRemoteV4.loadOrderStats()`. The passed value becomes really big when the date interval is very big, and then it triggers the error:

```
#0	0x000000010c180cd0 in OrderStatsRemoteV4.loadOrderStats(for:unit:timeZone:earliestDateToInclude:latestDateToInclude:quantity:forceRefresh:completion:) at /Users/hafiz/dev/ios/woocommerce-ios/Networking/Networking/Remote/OrderStatsRemoteV4.swift:45
#1	0x000000010aabee94 in StatsStoreV4.retrieveStats(siteID:timeRange:timeZone:earliestDateToInclude:latestDateToInclude:quantity:forceRefresh:onCompletion:) at /Users/hafiz/dev/ios/woocommerce-ios/Yosemite/Yosemite/Stores/StatsStoreV4.swift:144
#2	0x000000010aabdccc in StatsStoreV4.onAction(_:) at /Users/hafiz/dev/ios/woocommerce-ios/Yosemite/Yosemite/Stores/StatsStoreV4.swift:47
#3	0x000000010abf5590 in protocol witness for ActionsProcessor.onAction(_:) in conformance Store ()
#4	0x000000010a9cf0a8 in WeakProcessor.onAction(_:) at /Users/hafiz/dev/ios/woocommerce-ios/Yosemite/Yosemite/Base/Dispatcher.swift:104
#5	0x000000010a9ce880 in Dispatcher.dispatch(_:) at /Users/hafiz/dev/ios/woocommerce-ios/Yosemite/Yosemite/Base/Dispatcher.swift:76
#6	0x00000001012d3b9c in AuthenticatedState.onAction(_:) at /Users/hafiz/dev/ios/woocommerce-ios/WooCommerce/Classes/Yosemite/AuthenticatedState.swift:160
#7	0x00000001012d3cd0 in protocol witness for StoresManagerState.onAction(_:) in conformance AuthenticatedState ()
#8	0x00000001023023d0 in DefaultStoresManager.dispatch(_:) at /Users/hafiz/dev/ios/woocommerce-ios/WooCommerce/Classes/Yosemite/DefaultStoresManager.swift:140
#9	0x00000001023053dc in protocol witness for StoresManager.dispatch(_:) in conformance DefaultStoresManager ()
#10	0x0000000101792924 in DashboardViewModel.syncStats(for:siteTimezone:timeRange:latestDateToInclude:forceRefresh:onCompletion:) at /Users/hafiz/dev/ios/woocommerce-ios/WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift:120
#11	0x0000000101b7f968 in closure #1 in StoreStatsAndTopPerformersViewController.syncStats(forced:viewControllerToSync:onCompletion:) at /Users/hafiz/dev/ios/woocommerce-ios/WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsAndTopPerformersViewController.swift:233
#12	0x0000000101b854b8 in partial apply for closure #1 in StoreStatsAndTopPerformersViewController.syncStats(forced:viewControllerToSync:onCompletion:) ()
#13	0x0000000192a73d94 in Sequence.forEach(_:) ()
#14	0x0000000101b7eb6c in StoreStatsAndTopPerformersViewController.syncStats(forced:viewControllerToSync:onCompletion:) at /Users/hafiz/dev/ios/woocommerce-ios/WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsAndTopPerformersViewController.swift:198
#15	0x0000000101b7decc in StoreStatsAndTopPerformersViewController.syncAllStats(forced:onCompletion:) at /Users/hafiz/dev/ios/woocommerce-ios/WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsAndTopPerformersViewController.swift:161
#16	0x0000000101b7dde0 in closure #1 in StoreStatsAndTopPerformersViewController.reloadData(forced:) at /Users/hafiz/dev/ios/woocommerce-ios/WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsAndTopPerformersViewController.swift:129
#17	0x00000001e44fe084 in closure #1 in withCheckedContinuation<τ_0_0>(function:_:) ()
#18	0x00000001e44fe0bc in partial apply for closure #1 in withCheckedContinuation<τ_0_0>(function:_:) ()
#19	0x00000001e44fdef4 in withUnsafeContinuation<τ_0_0>(_:) ()
#20	0x00000001e44fde5c in withCheckedContinuation<τ_0_0>(function:_:) ()
#21	0x0000000101b7dbd0 in StoreStatsAndTopPerformersViewController.reloadData(forced:) at /Users/hafiz/dev/ios/woocommerce-ios/WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsAndTopPerformersViewController.swift:128
#22	0x00000001019dae40 in DashboardViewController.reloadDashboardUIStatsVersion(forced:) at /Users/hafiz/dev/ios/woocommerce-ios/WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift:430
#23	0x00000001019e221c in closure #2 in closure #1 in DashboardViewController.onPullToRefresh() at /Users/hafiz/dev/ios/woocommerce-ios/WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift:910
```

I looked into the Core code as best as I could but couldn't figure out what's the maximum number of accepted `per_page` parameter. Perhaps it's better to let the loading error happen, but:
1. Explain the issue (the date range is too wide)
2. Make it possible for users to choose a different date range

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by itsmeichigo** [comment]
FYI I found a comment [p3hLNG-Nm-p2#comment-4475] stating this:

> The earliest day that we have the raw data in our Hadoop cluster is 2015-10-01

~~This means we may need to validate the start date to be no earlier than the date above.~~
Update: We don't need to validate the start date as order and visitor stats are returned as 0 if the data is not available.

> I looked into the Core code as best as I could but couldn't figure out what's the maximum number of accepted per_page parameter. Perhaps it's better to let the loading error happen, but:

As mentioned in #12157, the maximum accepted `per_page` value is 100. But in that PR I set the default value for that param to always be 100, so I suppose it will fix  this issue. And the value should be sufficient without any implementation for pagination because we're setting fixed granularity [pe5sF9-2ri-p2#comment-3389] for now.

> Surprisingly, you can also tap on the empty space where the date range label is supposed to be, and it will show the date range picker.

I'll look into displaying the date range picker even when data cannot be displayed.
